lib/Dialect/FIRRTL/Transforms/ExpandWhens.cpp

```diff
@@ -551,24 +551,82 @@ class WhenOpVisitor : public LastConnectResolver<WhenOpVisitor> {
 
   /// Concurrent and of a property with the current condition.  If we are in
   /// the outer scope, i.e. not in a WhenOp region, then there is no condition.
-  Value ltlAndWithCondition(Operation *op, Value value) {
-    // 'ltl.and' the value with the current condition.
-    return OpBuilder(op).createOrFold<LTLAndIntrinsicOp>(
-        condition.getLoc(), condition.getType(), condition, value);
+  Value ltlAndWithCondition(Operation *op, Value property) {
+    // Look through nodes.
+    while (auto nodeOp = property.getDefiningOp<NodeOp>())
+      property = nodeOp.getInput();
+
+    // Look through `ltl.clock` ops.
+    if (auto clockOp = property.getDefiningOp<LTLClockIntrinsicOp>()) {
+      auto input = ltlAndWithCondition(op, clockOp.getInput());
+      auto &newClockOp = createdLTLClockOps[{clockOp, input}];
```

Comment:
why are you keeping track of these?

Reply:
This is just to reduce duplication. Since the condition is inserted _behind_ the clock, we basically have to copy the clock op since it might have other users that we can't modify. This has a tendency of creating redundant copies of the clock if a property is defined outside the `when` and reused for multiple asserts. The map here ensures that we only ever create a single copy per unique clock/input combination.